library/src/commonMain/kotlin/com/mercury/sqkon/db/QueryExt.kt

```diff
@@ -26,13 +37,13 @@ data class Eq<T : Any>(
 /**
  * Equivalent to `=` in SQL
  */
-infix fun <T : Any> JsonPathBuilder<T>.eq(value: String?): Eq<T> =
+infix fun <T : Any, V> JsonPathBuilder<T>.eq(value: V?): Eq<T, V> =
     Eq(builder = this, value = value)
 
 /**
  * Equivalent to `=` in SQL
  */
-inline infix fun <reified T : Any, reified V> KProperty1<T, V>.eq(value: String?): Eq<T> =
+inline infix fun <reified T : Any, reified V, VALUE> KProperty1<T, V>.eq(value: VALUE?): Eq<T, VALUE> =
     Eq(this.builder(), value)
 
 /**
```

Comment:
should all the rest of the functions (`notEq`, `inList`, etc) be genericized as well or is that a separate PR?

Reply:
Updated